utils/pg_notify.go

```diff
@@ -24,7 +24,7 @@ func ListenToPostgresNotify(pool *pgxpool.Pool, channel string, dbReconnectMaxDu
 		if err != nil {
```

Comment:
Shouldn't this file belong in the events package ? can't think of any usecases where it can be used as a generic utility

Reply:
This was moved to `utils` to avoid cyclic dependency.

package `playbook` needs it. And package `events` also needs `playbook`.

Reply:
@moshloop `pgnotifyrouter` is a very specific use case of PGNotify where we route based on the payload. This model doesn't fit well with all of our uses cases. Example: PGNotify on `playbook_updated` sends the updated playbook id as the payload.

I think, we can move the router to a new `events/pgnotifyrouter` but the `ListenToPostgresNotify` func in utils should stay.

Reply:
@adityathebe  do we not have the same problem for playbook_updated, we could have multiple background workers, but only 1 should receive the event ?

I also don't think utils should ever be used as a final package name - I would prefer moving it to utils/pg/notify.go and then the call becomes `pg.Listen(pool, channel)`  or better yet, `duty/pg/notify.go`

Reply:
> do we not have the same problem for playbook_updated, we could have multiple background workers, but only 1 should receive the event

We don't. We're still working with a single notify channel & only one of the workers will pick up the signal at a time.

> I also don't think utils should ever be used as a final package name - I would prefer moving it to utils/pg/notify.go and then the call becomes pg.Listen(pool, channel) or better yet, duty/pg/notify.go

Agreed 👍 will move to duty

Reply:
needed to add `Pool()` to our api.Context struct.
I've made a separate PR that makes this change https://github.com/flanksource/incident-commander/pull/566.